edgelet/provisioning/src/provisioning.rs

```diff
@@ -243,15 +281,20 @@ where
         let d = match c {
             Ok(c) => Either::A(
                 c.register()
-                    .map(|(device_id, hub_name)| {
+                    .map(|(device_id, hub_name, substatus)| {
                         info!(
                             "DPS registration assigned device \"{}\" in hub \"{}\"",
                             device_id, hub_name
                         );
+                        let reconfigure = substatus.map_or_else(
```

Comment:
It's probably better to make `DeviceRegistrationResult::substatus` have type `ReprovisioningStatus` instead of converting it ad-hoc here.

Basically make that field:

```rust
    #[serde(deserialize_with = "deserialize_reprovisioning_status")]
    substatus: ReprovisioningStatus,
```

and make a

```rust
fn deserialize_reprovisioning_status<'de, D>(deserializer: D) -> Result<ReprovisioningStatus, D::Error> where D: serde::Deserializer<'de> {
    let s = deserializer.deserialize_str()?;
    /* what is currently in the FromStr impl */
}
```

Or impl `serde::Deserialize` on `ReprovisioningStatus` manually in the same way.

Reply:
I think substatus could be use more generic for other cases in the future, it doesn't have to be directly linked to ReprovisioningStatus. Also model.rs is generated from DPS swagger, it has a lot of manual changes, but I think we should leave it as it was generated to make it easier to generate new versions.